rotkehlchen/db/updates.py

```diff
@@ -80,6 +82,45 @@ def _get_remote_info_json(self) -> dict[str, Any]:
 
         return json_data
 
+    def update_rpc_nodes(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        rpc nodes for those versions.
+        """
+        for version in range(local_version + 1, to_version + 1):
+            file_url = RPC_NODES_URL.format(branch=self.branch, version=version)
+            try:
+                updates = query_file(file_url, is_json=True)
+            except RemoteError as e:
+                log.warning(f'Failed to update rpc nodes due to {str(e)}')
+                break
+
+            updated_rpc_nodes = updates.get('rpc_nodes')
+            if updated_rpc_nodes is None:
+                log.error(f'Remote update {file_url} does not contain rpc_nodes key')
+                break
+
+            log.info(f'Applying update for rpc nodes from v{version}')
+            with (
+                GlobalDBHandler().conn.read_ctx() as cursor,
+                GlobalDBHandler().conn.write_ctx() as write_cursor,
+            ):
+                cursor.execute('DELETE FROM rpc_nodes')
```

Comment:
this should have been a delete cursor. **Important** This is very dangerous because if the user had custom rpc nodes you will delete them. Also you will delete the nodes for all the chains so that means that you will always need a full list of all the nodes that need to exists, then... why do you need the versioning?

What I would do is in the update file keep a list of nodes to add and a list of nodes to deprecate if they are present in the database taking the endpoint as identifier